src/Storage/FlysystemStorage.php

```diff
@@ -97,7 +97,12 @@ public function resolveUri(object|array $obj, ?string $fieldName = null, ?string
         $fs = $this->getFilesystem($mapping);
 
         try {
-            return $fs->publicUrl($path);
+            return $fs->publicUrl($path, [
```

Comment:
Could you elaborate a bit on this? I don't get why we need to pass our internal information (like the mapping and the field name) to Flysystem

Reply:
@garak sometimes, the url generator must be aware of which object and which fields is getting generated.

For example, an `User` entity with `profilePictureFile` (public) and `bankDetailFile` (private) fields.
The generator must generate a public url for one, and a private url with authorization for the other field.

Passing internals like mapping would not be needed in this case so it can be removed if you think it's not worth it